service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -0,0 +1,296 @@
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String buildReasonCodeDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasReasonCode() && (referralRequest.getReasonCode().size() > 1)) {
+            return REASON_CODE + extractReasonCodeString(referralRequest);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String extractReasonCodeString(ReferralRequest referralRequest) {
+        var ignoreFirstReasonCode = referralRequest.getReasonCode().subList(1, referralRequest.getReasonCode().size());
+        return ignoreFirstReasonCode.stream()
+            .map(value -> extractTextOrCoding(value))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(Collectors.joining(COMMA));
+    }
+
+    private String buildNoteDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasNote()) {
+            return referralRequest.getNote().stream()
+                .map(value -> String.format(NOTE, getAuthorString(value), getNoteTime(value),  value.getText()))
+                .collect(Collectors.joining(COMMA));
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String getAuthorString(Annotation annotation) {
+        if (annotation.hasAuthorStringType() && annotation.getAuthorStringType().hasValue()) {
+            return NOTE_AUTHOR + annotation.getAuthorStringType().getValue();
+        } else if (annotation.hasAuthorReference() && annotation.getAuthorReference().hasReferenceElement()) {
+            IIdType reference = annotation.getAuthorReference().getReferenceElement();
+            if (reference.getResourceType().equals(ResourceType.RelatedPerson.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (RelatedPerson) resource)
+                    .map(value -> NOTE_AUTHOR_RELATION + getHumanNameString(value.getNameFirstRep()))
+                    .orElse(StringUtils.EMPTY);
+            } else if (reference.getResourceType().equals(ResourceType.Practitioner.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (Practitioner) resource)
+                    .map(value -> NOTE_AUTHOR_PRACTITIONER + getHumanNameString(value.getNameFirstRep()))
+                    .orElse(StringUtils.EMPTY);
+            } else if (reference.getResourceType().equals(ResourceType.Patient.name())) {
+                return NOTE_AUTHOR_PATIENT;
+            }
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private String getHumanNameString(HumanName humanName) {
+        if  (humanName.hasText()) {
+            return humanName.getText();
+        } else if (humanName.hasGiven() || humanName.hasFamily()) {
+            String given = Optional.ofNullable(humanName.getGivenAsSingleString()).orElse(StringUtils.EMPTY);
```

Comment:
Wrapping variables in optionals for the sake of using "orElse" is not something which we should do. How about using a string builder and append text if present?